packages/eslint/types/index.d.ts

```diff
@@ -37,7 +37,7 @@ export interface RollupEslintOptions extends ESLint.Options {
    * Custom error formatter or the name of a built-in formatter.
    * @default stylish
    */
-  formatter?: CLIEngine.Formatter | string;
+  formatter?: Awaited<ReturnType<ESLint['loadFormatter']>>['format'] | string;
```

Comment:
the docs currently say that a regular `Function` is valid here, and it still technically is even with `await`. so the docs need to be updated to reflect this change. docs should also reflect that the return value of the function should be `String | Promise<String>`

Given that the last update to the plugin missed the broken types, it would be great (but not required for this PR) to add type tests (other plugins have these) or to update the tests to use TypeScript. again, not required for approval but would be a big improvement.